src/chart.zoom.js

```diff
@@ -19,185 +19,222 @@ var panFunctions = zoomNS.panFunctions = zoomNS.panFunctions || {};
 
 // Default options if none are provided
 var defaultOptions = zoomNS.defaults = {
-	pan: {
-		enabled: true,
-		mode: 'xy',
-        speed: 20,
-		threshold: 10,
-	},
-	zoom: {
-		enabled: true,
-		mode: 'xy',
-	}
+        pan: {
+                enabled: true,
+                mode: 'xy',
+                speed: 20,
+                threshold: 10,
+        },
+        zoom: {
+                enabled: true,
+                mode: 'xy',
+                sensitivity: 3,
+        }
 };
 
 function directionEnabled(mode, dir) {
-	if (mode === undefined) {
-		return true;
-	} else if (typeof mode === 'string') {
-		return mode.indexOf(dir) !== -1;
-	}
+        if (mode === undefined) {
+                return true;
+        } else if (typeof mode === 'string') {
+                return mode.indexOf(dir) !== -1;
+        }
 
-	return false;
+        return false;
 }
 
-function zoomIndexScale(scale, zoom, center) {
-
+function zoomIndexScale(scale, zoom, center, zoomOptions) {
+        var labels = scale.chart.data.labels;
+        var minIndex = scale.minIndex;
+        var lastLabelIndex = labels.length - 1;
+        var maxIndex = scale.maxIndex;
+        var sensitivity = zoomOptions.sensitivity;
+
+        zoomNS.zoomCumulativeDelta = zoom > 1 ? zoomNS.zoomCumulativeDelta + 1 : zoomNS.zoomCumulativeDelta - 1;
+
+        if (Math.abs(zoomNS.zoomCumulativeDelta) > sensitivity){
+                if(zoomNS.zoomCumulativeDelta < 0){
+                        if(center.x <= scale.width/2){
+                                if (minIndex <= 0){
+                                        maxIndex = Math.min(lastLabelIndex, maxIndex + 1);
+                                } else{
+                                        minIndex = Math.max(0, minIndex - 1);
+                                }
+                        } else if(center.x > scale.width/2){
+                                if (maxIndex >= lastLabelIndex){
+                                        minIndex = Math.max(0, minIndex - 1);
+                                } else{
+                                        maxIndex = Math.min(lastLabelIndex, maxIndex + 1);
+                                }
+                        }
+                        zoomNS.zoomCumulativeDelta = 0;
+                }
+                if(zoomNS.zoomCumulativeDelta > 0){
```

Comment:
Minor point:
I think this should be `else if (zoomNS.zoomCumulativeDelta > 0)` unless I've missed a way that the cumulative delta can become negative in the first block